src/host/os_linkdir.c

```diff
@@ -46,8 +46,27 @@ int do_linkdir(lua_State* L, const char* src, const char* dst)
 		return res != 0;
 	}
 #else
-    int res = symlink(src, dst);
-    return res == 0;
+	(void)L;
+	if (!do_isabsolute(src))
+	{
+		char cwd[PATH_MAX];
+		if (!do_getcwd(cwd, PATH_MAX))
+		{
+			return FALSE;
+		}
+
+		char relSrcPath[2 * PATH_MAX + 1];
+		snprintf(relSrcPath, 2 * PATH_MAX + 1, "%s/%s", cwd, src);
+		relSrcPath[PATH_MAX - 1] = '\0';
```

Comment:
Is it normal to have `2 * PATH_MAX + 1` and use `PATH_MAX - 1` here?

Apply to below code too.

Reply:
That is incorrect. Will fix.